### PR TITLE
feat(diffusion): add MiniMax H3 AdaLN cache

### PR DESCRIPTION
**Pull request by mickqian**
## Summary

- add an explicit MiniMax H3 AdaLN sidecar-cache format and CUDA builder
- skip original AdaLN checkpoint keys only when a matching cache is selected
- reject timestep embeddings absent from the cache rather than changing the inference path
- document the opt-in CLI and cookbook workflow

## Motivation

The MiniMax H3 model card notes that its AdaLN branches can be precomputed for inference-only serving. The public checkpoint ships those original branches, not a cache artifact, so this keeps the existing path as the default and makes cache use explicit.

## Status

Draft: remote end-to-end numerical and peak-memory validation are pending.

---
### CI States

Latest PR Test (Base): :x: [Run #31182344526](https://github.com/sgl-project/sglang/actions/runs/31182344526)
Latest PR Test (Extra): :x: [Run #31182344502](https://github.com/sgl-project/sglang/actions/runs/31182344502)

### CODE REVIEW

**Comment by triple-mu** [comment]
Two test findings from building on this branch. I do not have push access here, so patches inline.

**1. `test_minimax_h3_adaln_cache_matches_bf16_embedding` fails on this branch today.**

The implementation is correct; the assertion compares the wrong shape. `block()` reshapes a plan's `[num_timesteps, 6 * modality * hidden]` row into six `[num_timesteps * modality, hidden]` chunks -- the modality axis folds into the leading one -- so the concatenation is `(6, 24)` while `block_params[1, :, 1]` is `(2, 72)`. Same elements in the same flat order, so `torch.equal` returns False on the shape alone. `(6, 24)` is the layout the forward pass needs, matching what `MiniMaxH3AdalnProj.split_output` produces on the resident-weight path.

```diff
-    assert torch.equal(torch.cat(block, dim=-1), block_params[1, :, 1])
+    assert torch.equal(
+        torch.cat(block, dim=-1).reshape(block_params[1, :, 1].shape),
+        block_params[1, :, 1],
+    )
```

Line 52 (`final`) is fine as-is -- `final_width` has no modality axis, so its leading dim is unchanged.

**2. A rebase to current main will break `test_direct_device_loading_skips_rank_local_cpu_checkpoint`.**

Not an issue on this branch, since that test does not exist at your base (`4020bc95a7`). But it landed upstream since, and it asserts the old `safetensors_weights_iterator` signature, which this PR extends with `key_filter`:

```diff
     weight_iterator.assert_called_once_with(
         ["model.safetensors"],
+        key_filter=None,
         weight_load_plan=load_plan,
     )
```

Also worth flagging for that rebase: this PR threads `key_filter` into one `safetensors_weights_iterator` call site in `fsdp_load.py`, but current main has two -- the `weight_load_plan` branch was added since. Missing the second one silently disables the filter on the `load_full_state_dict_on_device` path, and the adaLN weights get loaded anyway.

Context: I rebased this branch plus #34575 and #34650 onto `26579d893` and ran them on 8x RTX PRO 5000. The AdaLN cache path is holding up well -- products are byte-identical to the resident-weight path across t2va/fl2va/ref2va at both tp1/ul8 and tp2/ul4, and it is what lets tp1 keep the whole DiT resident (127.03 s vs 135.74 s for the best layerwise-offload configuration).

**Comment by mickqian** [comment]
closed since absorbed in #34650
